File: THCS/2022_2023/ThanhHoa_2223/Bai2/TANGQUA.py
def main():
    # Đọc dữ liệu từ file
    with open("TANGQUA.INP", "r") as infile:
        N = int(infile.readline().strip())  # Số lượng món quà
        gifts = list(map(int, infile.readline().strip().split()))  # Danh sách giá trị quà

    # Tìm giá trị lớn nhất trong danh sách quà
    max_value = max(gifts)
    
    # Đếm số lượng quà có giá trị lớn nhất
    count_max_gifts = gifts.count(max_value)

    # Ghi kết quả vào file
    with open("TANGQUA.OUT", "w") as outfile:
        outfile.write(str(count_max_gifts) + "\n")

if __name__ == "__main__":
    main()

File: THCS/2022_2023/ThanhHoa_2223/Bai2/TANGQUA.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    // Đọc dữ liệu từ file
    ifstream infile("TANGQUA.INP");
    int N;
    infile >> N; // Số lượng món quà
    vector<int> gifts(N);

    // Đọc giá trị quà
    for (int i = 0; i < N; ++i) {
        infile >> gifts[i];
    }
    infile.close();

    // Tìm giá trị lớn nhất
    int max_value = *max_element(gifts.begin(), gifts.end());

    // Đếm số lượng quà có giá trị lớn nhất
    int count_max_gifts = count(gifts.begin(), gifts.end(), max_value);

    // Ghi kết quả vào file
    ofstream outfile("TANGQUA.OUT");
    outfile << count_max_gifts << endl;
    outfile.close();

    return 0;
}
